### ISSUE TITLE
🐛 Login server - Incorrect Source IP's?

### ISSUE THREAD

**Issue by CatsEyeXI** [DerpyProjectGroup/topaz#3537]
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my issue will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.

## OS / platform the server is running (if known)

Win10

## Branch affected by issue 

`base`

## Steps to reproduce

This issue has been going on for several months but I've held off on raising an issue as I know a re-write is in progress.
However, I'm unable to connect to my own server (on my LAN, (CatsEye)).

The issue I'm seeing, is that I'm seeing a ton of `shutdown_socket` messages for 172.16.99.1. This IP address is the IP of my gateway (router). It is common to both my desktop and the server. However, I have **zero** existing connections to the server, i.e. no characters logged in. Ultimately you can see where it says "too many connection attempts"... This doesn't make sense... That IP is not assigned to a device that would be trying to connect.

Please let me know if there is additional info I can help provide. I can reproduce this issue reliably. i.e. If you need packet captures.

![image](https://user-images.githubusercontent.com/65316697/215088867-755a3cbb-84a8-47d0-a87a-f5d145cf12c1.png)

## Expected behavior

Should be able to connect to server

**Comment by RAIST5150** [comment]
Wonder if their is something funky going on with NAT loopback/hairpinning? Or does it persist even if a client on the local network isn't trying to connect?

**Comment by CatsEyeXI** [comment]
> Wonder if their is something funky going on with NAT loopback/hairpinning? Or does it persist even if a client on the local network isn't trying to connect?

those connections aren't from me or my network, i have no idea why its using 172.16.99.1 as the IP.. i'm not even trying to connect.

**Comment by RAIST5150** [comment]
It is in one of the reserved blocks for private use, which is why I thought it might be on your local network.

¯\_(ツ)_/¯

**Comment by CatsEyeXI** [comment]
> It is in one of the reserved blocks for private use, which is why I thought it might be on your local network.
> 
> 
> 
> ¯\_(ツ)_/¯ 

Yeah. Am aware. That's my Vlan's address space. Like I said, without any connection attempts from within my network going on. The login server is still getting the gateway ip for these sessions

**Comment by CatsEyeXI** [comment]
Check out the last line. Couldn't even pull an account name. 

![image](https://user-images.githubusercontent.com/65316697/215251700-b1e1635d-f32a-450f-9316-9328c62c216b.png)

**Comment by atom0s** [comment]
Do you have your website set up to try and open a socket to the login server to check if its online? This was an older means of detecting if it was online to show on sites. If that isn't cached it can result in the `too many connections` issue based on your site's traffic.

**Comment by CatsEyeXI** [comment]
> Do you have your website set up to try and open a socket to the login server to check if its online? This was an older means of detecting if it was online to show on sites. If that isn't cached it can result in the `too many connections` issue based on your site's traffic.

the website does check it in this manner... its a good call, strange though that its showing the source IP as my router? what is a possible workaround?

**Comment by CatsEyeXI** [comment]
i guess that NAT'ing thats going on *could* expose itself as `172.16.99.1` to the login server since its all on the same network... its it possible to add an exclusion to the login server somewhere? there's no account id associated with it so the normal means of adding it to the `ip_exceptions` table wouldn't work here.
